File: pm_core/src/core/domain/entity/workflow_transition_entity.go
/*
Author: QuanTuanHuy
Description: Part of Serp Project
*/

package entity

import "github.com/serp/pm-core/src/core/domain/enum"

type WorkflowTransitionEntity struct {
	BaseEntity

	WorkflowID  int64  `json:"workflowId"`
	FromStateID int64  `json:"fromStateId"`
	ToStateID   int64  `json:"toStateId"`
	Name        string `json:"name"`

	ActiveStatus string `json:"activeStatus"`
}

func NewWorkflowTransitionEntity() *WorkflowTransitionEntity {
	return &WorkflowTransitionEntity{
		ActiveStatus: string(enum.Active),
	}
}

func (t *WorkflowTransitionEntity) MatchesTransition(fromStateID, toStateID int64) bool {
	return t.FromStateID == fromStateID && t.ToStateID == toStateID
}

File: pm_core/src/main/java/serp/project/pmcore/core/domain/entity/WorkflowTransitionEntity.java
/**
 * Author: QuanTuanHuy
 * Description: Part of Serp Project
 */

package serp.project.pmcore.core.domain.entity;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
public class WorkflowTransitionEntity extends BaseEntity {
    private Long tenantId;
    private Long workflowId;
    private String name;
    private Long fromStatusId;
    private Long toStatusId;
    private Integer sequence;

    private List<WorkflowTransitionRuleEntity> rules;
}
